CMakeLists.txt

```diff
@@ -0,0 +1,303 @@
+		)
+
+		# Check unwind
+		check_include_files("unwind.h" UNWIND_HEADER_FOUND)
+		check_function_exists(
+			"_Unwind_Backtrace" UNWIND_FN_FOUND CMAKE_REQUIRED_INCLUDES
+		)
+		if((${UNWIND_HEADER_FOUND}) AND (${UNWIND_FN_FOUND}))
+			set(sources "${sources};backtrace.c;simple.c")
+		else()
+			set(sources "${sources};nounwind.c")
+			set(BACKTRACE_SUPPORTED 0)
+		endif()
+
+		# Check UnwindGetIPInfo
+		set(oldFlags ${CMAKE_REQUIRED_FLAGS})
+		set(CMAKE_REQUIRED_FLAGS "-Werror-implicit-function-declaration")
+		check_c_source_compiles("
+			#include \"unwind.h\"
+
+			struct _Unwind_Context *context;
+			int ip_before_insn = 0;
+
+			int main(void) {
+				return _Unwind_GetIPInfo (context, &ip_before_insn);
+			}
+		" HAVE_GETIPINFO)
+		set(CMAKE_REQUIRED_FLAGS ${oldFlags})
+
+		set(oldFlags ${CMAKE_REQUIRED_FLAGS})
+		set(CMAKE_REQUIRED_FLAGS "-funwind-tables")
+		CHECK_C_COMPILER_FLAG("" unwind_tables_supported)
+		if(unwind_tables_supported)
+			set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -funwind-tables")
+		endif()
+		set(CMAKE_REQUIRED_FLAGS ${oldFlags})
+
+		# Check threads
+		# TODO configure.ac checks hppa*-*-hpux* stuff, dunno what it is
+		check_c_source_compiles("
+			int i;
+			int main(void) {
+				__sync_bool_compare_and_swap (&i, i, i);
+				__sync_lock_test_and_set (&i, 1);
+				__sync_lock_release (&i);
+				return 0;
+			}
+		" HAVE_SYNC_FUNCTIONS)
+		if(${HAVE_SYNC_FUNCTIONS})
+			set(BACKTRACE_SUPPORTS_THREADS 1)
+		endif()
+
+		# Check __atomic support
+		check_c_source_compiles("
+			int i;
+			int main(void) {
+				__atomic_load_n (&i, __ATOMIC_ACQUIRE);
+				__atomic_store_n (&i, 1, __ATOMIC_RELEASE);
+				return 0;
+			}
+		" HAVE_ATOMIC_FUNCTIONS)
+
+		# The library needs to be able to read the executable itself.
+		# Determine executable format (elf32/elf64/pecoff)
+		if(("${CMAKE_EXECUTABLE_FORMAT}" STREQUAL "ELF"))
+			math(EXPR BACKTRACE_ELF_SIZE "${CMAKE_SIZEOF_VOID_P} * 8")
+			set(filetype "elf${BACKTRACE_ELF_SIZE}")
+			set(FORMAT_FILE "elf.c")
+			set(BACKTRACE_SUPPORTS_DATA 1)
+		else()
+			# TODO detect pecoff/xcoff in some way - especially on MinGW
+			message(
+				FATAL_ERROR "Unknown executable format: '${CMAKE_EXECUTABLE_FORMAT}'"
+			)
+		endif()
+		message(STATUS "Executable format: ${filetype}")
+
+		# mmap support
+		check_include_files("sys/mman.h" HAVE_SYS_MMAN_H)
+		if(HAVE_SYS_MMAN_H)
+			# TODO port from configure.ac: spu-*-*|*-*-msdosdjgpp
+			check_function_exists(
+				"mmap" have_mmap CMAKE_REQUIRED_INCLUDES "sys/mman.h"
+			)
+		endif()
+		if(NOT have_mmap)
+			set(VIEW_FILE "read.c")
+			set(ALLOC_FILE "alloc.c")
+		else()
+			set(VIEW_FILE "mmapio.c")
+			check_c_source_compiles("
+				#include <sys/mman.h>
+				#if !defined(MAP_ANONYMOUS) && !defined(MAP_ANON)
+					#error no MAP_ANONYMOUS
+				#endif
+			" use_mmap_for_alloc)
+			if(use_mmap_for_alloc)
+				set(ALLOC_FILE "mmap.c")
+			else()
+				set(ALLOC_FILE "alloc.c")
+			endif()
+			if(${ALLOC_FILE} STREQUAL "alloc.c")
```

Comment:
no need to use ${} here, cmake will automatically unwrap the variable.

But you'd be better off moving the two SET() calls into the if(use_mmap_for_alloc) check anyway.